File: jphp-javafx-ext/src/main/java/org/develnext/jphp/ext/javafx/classes/UXParent.java
package org.develnext.jphp.ext.javafx.classes;

import javafx.scene.Parent;
import org.develnext.jphp.ext.javafx.JavaFXExtension;
import php.runtime.annotation.Reflection.Abstract;
import php.runtime.annotation.Reflection.Name;
import php.runtime.env.Environment;
import php.runtime.reflection.ClassEntity;

@Abstract
@Name(JavaFXExtension.NAMESPACE + "UXParent")
public class UXParent extends UXNode {
    interface WrappedInterface {
        void layout();
        void requestLayout();
    }

    public UXParent(Environment env, Parent wrappedObject) {
        super(env, wrappedObject);
    }

    public UXParent(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Override
    public Parent getWrappedObject() {
        return (Parent) super.getWrappedObject();
    }
}

File: jphp-javafx-ext/src/main/resources/JPHP-INF/sdk/ext/javafx/UXParent.php
<?php

namespace ext\javafx;

/**
 * Class UXParent
 * @package ext\javafx
 */
abstract class UXParent extends UXNode
{
    /**
     * Executes a top-down layout pass on the scene graph under this parent.
     */
    public function layout() {}

    /**
     * Requests a layout pass to be performed before the next scene is rendered.
     */
    public function requestLayout() {}

    /**
     * @return UXNode[]
     */
    public function getChildren() {}
}